lib/ash/type/type.ex

```diff
@@ -812,31 +812,31 @@ defmodule Ash.Type do
         term
         |> Enum.with_index()
         |> Enum.reduce({[], []}, fn {item, index}, {items, errors} ->
-          if is_nil(item) && not nil_items? do
-            {[item | items], [[message: "no nil values", index: index] | errors]}
+          if type.custom_apply_constraints_array?() do
+            {[item | items], errors}
```

Comment:
We will need to add a check here for `item` being `nil`, to match the behavior below (as you pointed out)